packages/base-cli/src/index.ts

```diff
@@ -0,0 +1,212 @@
+
+			process.exit(0);
+		}).catch((error) => {
+			console.error(argo.debug ? error : error.message);
+
+			process.exit(1);
+		});
+	}
+
+	if (argo.stdin) {
+		let output = argo.output;
+		if (!output && argo.outputDir) {
+			output = path.join(argo.outputDir, 'output.css');
+		}
+
+		return getStdin().then((css) => {
+			if (!css) {
+				helpLogger();
+				process.exit(1);
+			}
+
+			const result = postcss([plugin(argo.pluginOptions)]).process(css, {
+				from: 'stdin',
+				to: output,
+				map: (argo.inlineMap || argo.externalMap) ? { inline: argo.inlineMap } : false,
+			});
+
+			if (argo.externalMap && result.map) {
+				return Promise.all([
+					writeFile(output, result.css),
+					writeFile(`${output}.map`, result.map.toString()),
+				]).then(() => {
+					console.log(`CSS was written to "${path.normalize(output)}"`);
+					return;
+				});
+			}
+
+			return writeFile(output, result.css + (argo.inlineMap ? '\n' : ''));
+		}).then(() => {
+			console.log(`CSS was written to "${path.normalize(output)}"`);
+
+			process.exit(0);
+		}).catch((error) => {
+			console.error(argo.debug ? error : error.message);
+
+			process.exit(1);
+		});
+	}
+
+	if (argo.stdout) {
+		const outputs = argo.inputs.map((input) => {
+			return {
+				input: input,
+				result: null,
+			};
+		});
+
+		return Promise.all(argo.inputs.map((input) => {
+			return readFile(input).then((css) => {
+				const result = postcss([plugin(argo.pluginOptions)]).process(css, {
+					from: input,
+					to: 'stdout',
+					map: false,
+				});
+
+				return result.css;
+			}).then((result) => {
+				outputs.find((output) => output.input === input).result = result;
+			});
+		})).then(() => {
+			outputs.forEach((output) => {
+				process.stdout.write(output.result);
+			});
+
+			process.exit(0);
+		}).catch((error) => {
+			console.error(argo.debug ? error : error.message);
+
+			process.exit(1);
+		});
+	}
+
+	return Promise.all(argo.inputs.map((input) => {
+		let output = argo.output;
+		if (argo.outputDir) {
+			output = path.join(argo.outputDir, path.basename(input));
+		}
+		if (argo.replace) {
+			output = input;
+		}
+
+		return readFile(input).then((css) => {
+			return postcss([plugin(argo.pluginOptions)]).process(css, {
+				from: input,
+				to: output,
+				map: (argo.inlineMap || argo.externalMap) ? { inline: argo.inlineMap } : false,
+			});
+		}).then((result) => {
+			if (argo.externalMap && result.map) {
+				return Promise.all([
+					writeFile(output, result.css),
+					writeFile(`${output}.map`, result.map.toString()),
+				]).then(() => {
+					console.log(`CSS was written to "${path.normalize(output)}"`);
+					return;
+				});
+			}
+
+			return writeFile(output, result.css + (argo.inlineMap ? '\n' : '')).then(() => {
+				console.log(`CSS was written to "${path.normalize(output)}"`);
+			});
+		});
+	})).catch((error) => {
+		console.error(argo.debug ? error : error.message);
+
+		process.exit(1);
+	});
```

Comment:
I am not really happy that we have all this duplication here.

But each case is very different and splitting it this way was easiest to have no bugs.